src/formatDate.js

```diff
@@ -7,8 +7,34 @@
  *
  * @returns {string}
  */
+
+const getObjectDate = (dateParts, fromFormat) =>
+  fromFormat.reduce((objectFormat, format, i) => {
+    if (format === 'YY' && dateParts[i].length === 2) {
+      objectFormat['YYYY'] =
+        dateParts[i] < '30' ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    } else {
+      objectFormat[format] = dateParts[i];
+    }
+
+    return objectFormat;
+  }, {});
+
+const convertYear = (objectDate, toFormat) => {
+  if (!objectDate['YY'] && objectDate['YYYY'] && toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(-2);
+  }
```

Comment:
The condition here checks for the absence of 'YY' in `objectDate` but does not check if 'YYYY' is present before slicing it. This could potentially lead to errors if 'YYYY' is not a property on `objectDate`. It's better to ensure 'YYYY' exists before attempting to access its value.